pyproject.toml

```diff
@@ -47,6 +45,8 @@ test = ["pytest", "tox"]
 mainapp = [
     "gunicorn",
     "gevent",
+    'greenlet < 3.0; python_version<"3.12"',
+    'greenlet >= 3.0.0a1; python_version>="3.12.0rc0"',
```

Comment:
this was a workeround needed before greenlet 3.0 was released. gevent depends on greenlet only when needed so it shouldn't be here at all

```suggestion
```